lib/parser.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * Includes revisions to the block parser not yet proposed for core adoption.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Filters to override default block parser with Gutenberg's custom
+ * implementation.
+ *
+ * @since 6.9.0
+ */
+function gutenberg_replace_block_parser_class() {
+	return 'WP_Sourced_Attributes_Block_Parser';
+}
+add_filter( 'block_parser_class', 'gutenberg_replace_block_parser_class' );
```

Comment:
this is how the system was designed to work, though I believe that you should be able to skip creating the helper function.

```php
add_filter( 'block_parser_class', 'WP_Sourced_Attributes_Block_Parser' );
```

Reply:
This helper is useful, as it allows for this filter to be unhooked.

Reply:
>this is how the system was designed to work, though I believe that you should be able to skip creating the helper function.

I'd tend to agree with @spacedmonkey here, though it certainly piques my interest that this syntax can work 🤔